modules/plugins/languages/hcl.nix

```diff
@@ -0,0 +1,116 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (builtins) attrNames;
+  inherit (lib.options) mkEnableOption mkOption;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.types) package bool enum;
+  inherit (lib.nvim.types) mkGrammarOption;
+
+  cfg = config.vim.languages.hcl;
+
+  defaultServer = "terraform-ls";
+  servers = {
+    terraform-ls = rec {
+      package = pkgs.terraform-ls;
+      lspConfig = ''
+        lspconfig.terraformls.setup {
+          capabilities = capabilities,
+          on_attach=default_on_attach,
+          cmd = {"${lib.getExe package}", "serve"},
+        }
+      '';
+    };
+  };
+
+  defaultFormat = "hclfmt";
+  formats = {
+    hclfmt = rec {
+      package = pkgs.hclfmt;
+      nullConfig = ''
+        table.insert(
+          ls_sources,
+          null_ls.builtins.formatting.hclfmt.with({
+            command = "${lib.getExe package}",
+          })
+        )
+      '';
+    };
+  };
+in {
+  options.vim.languages.hcl = {
+    enable = mkEnableOption "HCL support";
+
+    treesitter = {
+      enable = mkEnableOption "HCL treesitter" // {default = config.vim.languages.enableTreesitter;};
+      package = mkGrammarOption pkgs "hcl";
+    };
+
+    lsp = {
+      enable = mkEnableOption "HCL LSP support (terraform-ls)" // {default = config.vim.languages.enableLSP;};
+      # TODO: (maybe, is it better?) it would be cooler to use vscode-extensions.hashicorp.hcl probably, shouldn't be too hard
+      package = mkOption {
+        description = "HCL language server package (terraform-ls)";
+        type = package;
+        default = servers.${defaultServer}.package;
+      };
+    };
+
+    format = {
+      enable = mkOption {
+        description = "Enable HCL formatting";
+        type = bool;
+        default = config.vim.languages.enableFormat;
+      };
+      type = mkOption {
+        description = "HCL formatter to use";
+        type = enum (attrNames formats);
+        default = defaultFormat;
+      };
+      package = mkOption {
+        description = "HCL formatter package";
+        type = package;
+        default = formats.${cfg.format.type}.package;
+      };
```

Comment:
Could you move description to the bottom? This is mostly a personal nit, but `description` is the option with most potential to span over multiple lines so it's better to have it at the bottom of `mkOption` args.